src/main/java/uk/gov/hmcts/ccd/domain/service/getcase/GetCaseCallback.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.ccd.domain.service.getcase;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewField;
+import uk.gov.hmcts.ccd.domain.model.callbacks.GetCaseCallbackResponse;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.service.stdapi.CallbackInvoker;
+import uk.gov.hmcts.ccd.endpoint.exceptions.CallbackException;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class GetCaseCallback {
+    private static final Logger LOG = LoggerFactory.getLogger(GetCaseCallback.class);
+
+    private final CallbackInvoker callbackInvoker;
+
+    public GetCaseCallback(CallbackInvoker callbackInvoker) {
+        this.callbackInvoker = callbackInvoker;
+    }
+
+    public GetCaseCallbackResponse invoke(final CaseTypeDefinition caseTypeDefinition, final CaseDetails caseDetails,
+                                          final List<CaseViewField> metadataFields) {
+        GetCaseCallbackResponse response;
+        try {
+            ResponseEntity<GetCaseCallbackResponse> getCaseCallbackResponse = callbackInvoker
+                .invokeGetCaseCallback(caseTypeDefinition, caseDetails);
+
+            response = getCaseCallbackResponse.getBody();
+        } catch (Exception e) {
+            LOG.error("CCD_CDI_CallbackGetCaseUrl: " + e.getMessage());
+            throw new CallbackException(e.getMessage());
+        }
+
+        validateNewMetadataFields(metadataFields, response);
+
+        return response;
+    }
+
+    private void validateNewMetadataFields(List<CaseViewField> metadataFields, GetCaseCallbackResponse response) {
+        if (response == null) {
+            throw new CallbackException("CCD_CDI_CallbackGetCaseUrl: response body not set");
+        }
+        Set<String> caseViewFieldIds = metadataFields.stream()
+            .map(CaseViewField::getId).collect(Collectors.toSet());
+        Set<String> callbackMetadataIds = response.getMetadataFields().stream()
+            .map(CaseViewField::getId).collect(Collectors.toSet());
+        callbackMetadataIds.retainAll(caseViewFieldIds);
```

Comment:
metadataFields are added into metadataFields, this might add duplicate values in 
https://github.com/hmcts/ccd-data-store-api/pull/1720/files#diff-8019fed3878686ae1a155dcfcefae36bab3bd01a2d11d549e44239395c61bfffR99

Reply:
It should not. This code is to throw the exception when duplicate id is found. Tested in: https://github.com/hmcts/ccd-data-store-api/pull/1720/files#diff-43a067b96d2bf84f64ca566a55694af8df36a20d83b0d202be559cebcf54debbR94
And in the IT:
https://github.com/hmcts/ccd-data-store-api/pull/1720/files#diff-f82753bcfb43bb9f34e1d3acc0c43b0a5d2a5cd75585d847bf3076bf6267faecR169